src/assets/asset_manager.rs

```diff
@@ -62,31 +67,35 @@ impl AssetManager {
             }
             if file_name.ends_with(".gltf") {
                 let current_index = self.materials.len() as u32;
-                let (mesh, materials) = Mesh::new(
-                    &device,
-                    format!("{}{}", full_file_path, file_name),
-                    current_index,
-                );
-                let mut index = current_index;
-                for material in materials {
-                    self.materials.insert(index, material);
-                    index += 1;
+                let gltf_data =
+                    GltfData::load(&device, format!("{}{}", full_file_path, file_name)).unwrap();
+                // TODO: Figure out what this needs to look like
+                let mesh = gltf_data.mesh;
+                for (handle,submeshes) in mesh.data {
+                    self.materials.insert(handle, None);
                 }
-                self.meshes.insert(file_name.to_string(), mesh);
+                self.meshes.entry(file_name.to_string()).or_insert( mesh);
                 info!("Loaded mesh: {}", file_name);
             }
-            if file_name.ends_with(".png")
-                || file_name.ends_with(".jpg")
-                || file_name.ends_with(".hdr")
-            {
-                let image = Image::new(
-                    &device,
-                    &mut init_encoder,
-                    format!("{}{}", full_file_path, file_name),
-                    file_name.to_string(),
-                );
-                self.images.insert(file_name.to_string(), image);
+            //TODO: Dont load pictures here
```

Comment:
Where would we load images instead?

Reply:
So when loading the material:
```  /// load_data loads the data specified in self if not present in images and returns them bundled as a materialdata
impl NewMaterialHandle{
    pub fn load_data(
        self,
        images: &mut HashMap<String,Arc<Image>>,
        device: &wgpu::Device,
        encoder: &mut wgpu::CommandEncoder,
    ) -> NewMaterialData {
   [...]
}
```
The images get loaded if they arent already.

Reply:
We still need to load some images that aren't loaded by the mesh though. An example of this would be the BRDF texture.

Reply:
Yea thats true. Could we include that in our material?
We can still load what we want during app load. Maybe even go for a loading phase. Right now the window is just unresponsive during load. That should be avoided I think.

Reply:
Ideally I want to re-work the asset manager to be a bit better. Some things I want to do differently. Have the ability to load assets across multiple threads. The ability to have `asset bundles`. Have some way for a user to detect and display something during load.